util/topsql/reporter/metrics.go

```diff
@@ -14,23 +14,50 @@
 
 package reporter
 
-import "github.com/pingcap/tidb/metrics"
+import (
+	"github.com/pingcap/tidb/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+)
 
 var (
-	ignoreExceedSQLCounter              = metrics.TopSQLIgnoredCounter.WithLabelValues("ignore_exceed_sql")
-	ignoreExceedPlanCounter             = metrics.TopSQLIgnoredCounter.WithLabelValues("ignore_exceed_plan")
-	ignoreCollectChannelFullCounter     = metrics.TopSQLIgnoredCounter.WithLabelValues("ignore_collect_channel_full")
+	ignoreExceedSQLCounter              prometheus.Counter
```

Comment:
It looks like most of the changes are such separation of declaration and initialization.
Do we have a strong motivation to deliver this change of code style?

Reply:
We will re-initialize metrics after getting `keyspace` from config, and these corresponding metrics variables should also be re-initialized. Thus, we conduct the separation of declaration and initialization.

Reply:
Can we just initialize metircs once?

Reply:
I'm not sure, it's worth a try. But no matter whether the metrics are initialized once or twice, the separation of declaration and initialization is needed.